hw18/api/api.spec.ts

```
@@ -0,0 +1,131 @@
+    const userId = 1337;
+    const postTitle = "postTitle";
+    const postBody = "postBody";
+
+    await superagent
+      .post(baseURL + "/posts")
+      .send({ userId: userId, title: postTitle, body: postBody })
+      .then(async (response) => {
+        expect(response.status).toBe(201);
+        expect(response.body.userId).toBe(userId);
+        expect(response.body.title).toBe(postTitle);
+        expect(response.body.body).toBe(postBody);
+        expect(response.body.id).toBe(101);
+      })
+      .catch((error) => error.message);
+  });
+
+  test("Should successfully update first post's title via PATCH method", async () => {
+    const updatedTitle = "postTitle";
+
+    await superagent
+      .patch(baseURL + "/posts/1")
+      .send({ title: updatedTitle })
+      .then(async (response) => {
+        expect(response.status).toBe(200);
+        expect(response.body.id).toBe(1);
+        expect(response.body.title).toBe(updatedTitle);
+        expect(response.body.userId).not.toBe(null);
+        expect(response.body.body).not.toBe(null);
+      })
+      .catch((error) => error.message);
+  });
+
+  test("Should successfully update post's title via PUT method", async () => {
+    const updatedTitle = "postTitle";
+
+    await superagent
+      .put(baseURL + "/posts/1")
+      .send({ title: updatedTitle })
+      .then(async (response) => {
+        expect(response.status).toBe(200);
+        expect(response.body.id).toBe(1);
+        expect(response.body.title).toBe(updatedTitle);
+        expect(response.body.userId).toBeUndefined();
+        expect(response.body.body).toBeUndefined();
+      })
+      .catch((error) => error.message);
+  });
+
+  test("Should successfully delete post", async () => {
+    await superagent
+      .delete(baseURL + "/posts/1")
+      .then(async (response) => {
+        expect(response.status).toBe(200);
+        expect(response.body).toEqual({});
+      })
+      .catch((error) => error.message);
```

Please throw an error, not just return it.